### ISSUE TITLE
Request for ARM build

### ISSUE THREAD

**Issue by wizonesolutions** [borsboom/foreign-currency-accounts-for-ynab#24]
Hi, I'm trying to get this set up on my Pi since I'm moving soon and don't want to keep it as a cronjob on an old Mac. I'm going to try building from source as per the instructions but figured I'd see if you had the capacity to add an ARM release. The x86_64 build naturally doesn't work.

**Comment by wizonesolutions** [comment]
Aww, seems like it won't work:

```
pi@raspberrypi:~/fca4ynab/src $ cargo install --path .
  Installing fca4ynab v0.1.10 (/home/pi/fca4ynab/src)
    Updating crates.io index
    Updating git repository `https://github.com/borsboom/ynab-api.git`
   Compiling semver-parser v0.7.0
   Compiling maybe-uninit v2.0.0
   Compiling libc v0.2.67
   Compiling autocfg v1.0.0
error: Could not compile `maybe-uninit`.

Caused by:
  process didn't exit successfully: `rustc --crate-name build_script_build /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/maybe-uninit-2.0.0/build.rs --color always --crate-type bin --emit=dep-info,link -C opt-level=3 -C metadata=27e1ba85f0efa35a -C extra-filename=-27e1ba85f0efa35a --out-dir /home/pi/fca4ynab/src/target/release/build/maybe-uninit-27e1ba85f0efa35a -L dependency=/home/pi/fca4ynab/src/target/release/deps --cap-lints allow` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
error: Could not compile `libc`.

Caused by:
  process didn't exit successfully: `rustc --crate-name build_script_build /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.67/build.rs --color always --crate-type bin --emit=dep-info,link -C opt-level=3 --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=14e6d7861e8782ee -C extra-filename=-14e6d7861e8782ee --out-dir /home/pi/fca4ynab/src/target/release/build/libc-14e6d7861e8782ee -L dependency=/home/pi/fca4ynab/src/target/release/deps --cap-lints allow` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
error: Could not compile `autocfg`.

Caused by:
  process didn't exit successfully: `rustc --crate-name autocfg /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/autocfg-1.0.0/src/lib.rs --color always --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C metadata=c8d680dfb8298080 -C extra-filename=-c8d680dfb8298080 --out-dir /home/pi/fca4ynab/src/target/release/deps -L dependency=/home/pi/fca4ynab/src/target/release/deps --cap-lints allow` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
thread '<unnamed>' panicked at 'src/librustc_codegen_ssa/back/write.rs:1545: worker thread panicked', src/librustc/util/bug.rs:37:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: failed to prepare thin LTO module: file doesn't start with bitcode header

error: aborting due to previous error

error: failed to compile `fca4ynab v0.1.10 (/home/pi/fca4ynab/src)`, intermediate artifacts can be found at `/home/pi/fca4ynab/src/target`

Caused by:
  Could not compile `semver-parser`.

Caused by:
  process didn't exit successfully: `rustc --crate-name semver_parser /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-parser-0.7.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C metadata=bef76ad9970293e1 -C extra-filename=-bef76ad9970293e1 --out-dir /home/pi/fca4ynab/src/target/release/deps -L dependency=/home/pi/fca4ynab/src/target/release/deps --cap-lints allow` (signal: 11, SIGSEGV: invalid memory reference)
```

Not sure if there's anything to really be done here. I will put it on a spare server instead for now. Happy to help test the ARM build, although it may take me a bit to get to it each time.

**Comment by borsboom** [comment]
Give this binary a try: http://downloads.borsboom.io/tmp/fca4ynab-0.1.10-armv7-linux

Note that I just cross-compiled it from Ubuntu x86, so I've not actually tested it to make sure it works since I don't have an ARM box handy.

**Comment by borsboom** [comment]
I've opened a PR that adds automated builds of ARMv7 binaries: #26.

You can grab a binary from https://dev.azure.com/borsboom/foreign-currency-accounts-for-ynab/_build/results?buildId=267&view=artifacts&type=publishedArtifacts

**Comment by borsboom** [comment]
Released in [v0.1.11](https://github.com/borsboom/foreign-currency-accounts-for-ynab/releases/tag/release/0.1.11).

**Comment by wizonesolutions** [comment]
Can confirm that this looks to be running fine on the Pi. I already have it set up on another server, so not actually using this yet since I'd have to get another free currency converter API key (one IP per key). But I think it will be useful in general, and I may yet switch to it at some point. Thanks!
